### PR TITLE
Add hyenaND-retrofit Claude skill

### PR DESCRIPTION
**Pull request by farhadrgh**
Adds a Claude Code skill that automates the attention -> HyenaND swap. Two paths:

- **Native**: in-repo configs that already use the `build_attention_net` / `build_hyena_net` / `build_hybrid_net` builders. Swap is mechanical — change the builder call, drop `compile_compatible_fftconv = False`, pick a layer pattern for hybrids. Sibling files are bare config shims (no `__main__` block, matching the canonical `hyena_patch16.py` / `hybrid_hhha.py`).
- **Foreign**: external PyTorch models (timm ViT, HF transformers, hand-written) using `nn.MultiheadAttention` or `F.scaled_dot_product_attention`. Skill emits a sibling file with a full Hyena module plus a drop-in `HyenaAttnAdapter` that bridges the three nn.MHA ↔ Hyena interface gaps: tuple return, kwarg surface (`need_weights`/`attn_mask`/`key_padding_mask`), and channel layout / CLS handling.

The foreign path is organized around four orthogonal axes the user fixes up front:

1. `data_dim ∈ {1, 2, 3}` — picks `ConvNd`, sets `CKConvND` / `SIRENKernelND` / `GaussianModulationND`.
2. `causal ∈ {True, False}` — sets `fft_padding`, `use_rope`, mask `parametrization`, `omega_0`.
3. Host layout — `tokens [B, N, C]` vs `feature_map [B, C, *S]`.
4. Return contract — `(out, None)` tuple vs bare tensor.

One parameterized adapter covers all sixteen combinations. FFT backend is selected automatically from the axes: `fft_backend="subq_ops"` (optimized CUDA kernel from `subquadratic_ops_torch`) when `data_dim == 2 and not causal`, else `fft_backend="torch_fft"` — the latter is required for 1D, 3D, circular padding, fp16, and causal paths because `subq_ops` is 2D-only in the current nvSubquadratic wiring (the upstream package ships `fft_causal_conv1d` and `causal_conv1d` but they are not exposed).

A foot-guns section calls out the failures that bite in practice: INT32 `im2col` overflow in large 3D short conv (≥160³ trips it; fix is `DepthwiseFFTConv3d`), RoPE divisibility (`% 2` / `% 4` / `% 6` for 1D / 2D / 3D, must hold at every hierarchical stage), `L_cache` memory blowup (`(2L-1)^D × D × 4` bytes — D=3 / L=512 ≈ 12.8 GB), state_dict-prefix mismatch on pretrained-attention loading, and the `subq_ops` constraint set (asserts loudly on `data_dim ≠ 2`, `fft_padding ≠ "zero"`, causal, or fp16).

Hierarchical encoders with a `use_hyena=True` flag are an explicit exception to the sibling-file rule — the natural API is an in-place edit.

Five eval cases under `evals/` cover the matrix:

- `native-pure-swap`, `native-hybrid` — v5_patch and vit5_hybrid configs.
- `foreign-timm-vit` — 2D, tokens, with CLS.
- `foreign-3d-feature-map` — 3D, feature_map, no CLS, `F.scaled_dot_product_attention` on `[B, C, D, H, W]`.
- `foreign-1d-causal-lm` — 1D, causal tokens, `attn_mask` call site.

Together they exercise every (`data_dim`, `causal`, `host_layout`, prefix-tokens) combination the skill is meant to handle.

### CODE REVIEW

File: .claude/skills/hyenand-retrofit/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+1. **`causal ∈ {True, False}`** — autoregressive 1D LMs are causal; vision, segmentation, PDE are bidirectional. Causal sets `is_causal=True` on `CKConvND` (1D only) plus `use_rope=True`, mask `parametrization="exp_decay"`, `omega_0=100`. Bidirectional sets `is_causal=False`, `use_rope=False`, `parametrization="direct"`, `omega_0=10`. Boundary condition (`fft_padding`) is a separate axis-4-style choice — see the FFT-backend section.
+
+1. **Host layout** — `tokens [B, N, C]` (most ViTs, causal LMs) or `feature_map [B, C, *spatial]` (CNNs, U-Nets, hierarchical encoders). Spatial dim count does *not* change this — a 1D, 2D, or 3D feature-map host all use `[B, C, *S] -> [B, *S, C]`.
+
+1. **Return contract** — `(out, None)` tuple if the call site is `h, _ = self.attn(...)` (matches `nn.MultiheadAttention`); bare tensor otherwise.
+
+## Pure or hybrid
+
+Separate decision: replace every attention site (pure) or leave some as attention (hybrid). Hybrids are common in vision and genomics LMs because attention's selectivity complements Hyena's global mixing. If unsure, ask via AskUserQuestion.
```
Comment (tshimko-nv) [inline .claude/skills/hyenand-retrofit/SKILL.md]: Is this skill intended only for Claude? If not, is AskUserQuestion universal to Codex, etc. as well?

**Review by tshimko-nv — APPROVED** [review]
Looks good - one small comment about Claude vs. universal skill implementation, but otherwise, should be good.
